src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -13,6 +13,8 @@
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
+    // TODO: Remove all optional parameters from IDiagnosticAnalyzerService
+    // Tracked with https://github.com/dotnet/roslyn/issues/67434
```

Comment:
I have added the new parameter as an optional parameter on existing methods that were taking all optional parameters. Moving the new parameter upfront seemed awkward, and we I plan to work on https://github.com/dotnet/roslyn/issues/67434 once this PR goes in, so it should become non-optional pretty soon after.

Reply:
wfm.